src/fabric/src/fabric.erl

```diff
@@ -185,6 +194,25 @@ set_security(DbName, SecObj) ->
 set_security(DbName, SecObj, Options) ->
     fabric_db_meta:set_security(dbname(DbName), SecObj, opts(Options)).
 
+%% @doc reset the time seq data structure
+-spec set_time_seq(dbname(), any()) -> ok.
```

Comment:
To avoid using `any()` here you can.

1. add `-opaque time_seq() :: couch_time_seq:time_seq().` in `couch/include/couch_db.hrl`
2. add following types in `couch_time_seq.erl`
```
-export_type([
    time_seq/0
]).

-type bin() :: {epoch(), update_seq()}.
-type time_seq() :: #{
     v => pos_integer(),
     bins => [bin()]
}.
```

You might need to define `update_seq()` and `epoch()` types as well (I assumed something like that exist already but I didn't check naming).

The `opaque` types are designed to avoid abstraction leaking and meant be used to define interfaces between different components.

Reply:
I added `-spec set_time_seq(dbname(), couch_time_seq:time_seq()) -> ok.` type annotations to `fabric`

Then ```-export_type([time_seq/0]).``` plus a few other suggested types to `couch_time_seq` module.
```